tests/integration/golang/helpers/s3.go

```diff
@@ -31,3 +35,51 @@ func NewS3Client(endpoint string) (*s3.Client, error) {
 		o.UsePathStyle = true
 	}), nil
 }
+
+func CreateBuckets(s3Client *s3.Client) error {
+	for _, bucket := range testBuckets {
+		_, err := s3Client.CreateBucket(context.Background(), &s3.CreateBucketInput{
+			Bucket:                    aws.String(bucket),
+			CreateBucketConfiguration: &types.CreateBucketConfiguration{},
+		})
+		if err != nil {
+			return fmt.Errorf("failed to create bucket '%s': %v", bucket, err)
+		}
+	}
+	return nil
+}
+
+func RemoveBuckets(s3Client *s3.Client) error {
+	for _, bucket := range testBuckets {
+		// Delete all objects in the bucket
+		listObjectsOutput, err := s3Client.ListObjectsV2(context.Background(), &s3.ListObjectsV2Input{
+			Bucket: aws.String(bucket),
+		})
+		if err != nil {
+			return fmt.Errorf("failed to list objects in bucket '%s': %v", bucket, err)
+		}
+		var objectIds []types.ObjectIdentifier
+		for _, object := range listObjectsOutput.Contents {
+			objectIds = append(objectIds, types.ObjectIdentifier{Key: aws.String(*object.Key)})
```

Comment:
this call to `aws.String` is totally unnecessary

Reply:
i think the sdk docs showed this usage

Reply:
Please also have a look at https://github.com/awsdocs/aws-doc-sdk-examples/blob/c2edcff1365d4b454b51075d632a1be844dd3e47/go/example_code/s3/s3_delete_bucket.go#L71 which shows that deleting a bucket is asynchronous and we thus need to check for completion.

Reply:
I don't think local Minio for integration tests, with with 1 item per bucket, requires this kind of bullet-proof approach

Reply:
AFAICT the `s3manager` used in the SDK examples you linked is not available in our V2 s3 packages, so I'd suggest we stick with "list-then-delete-objects" approach here using just the s3 client. 
```gwilson@macbook ~/P/f/t/i/g/m/artifact (integration-tests-devcontainer) [1]> go get github.com/aws/aws-sdk-go-v2/service/s3/s3manager
go: module github.com/aws/aws-sdk-go-v2/service/s3@upgrade found (v1.40.2), but does not contain package github.com/aws/aws-sdk-go-v2/service/s3/s3manager
```
I can add pagination on list and wait for delete.

Reply:
done